addons/mrp/models/mrp_abstract_workorder.py

```diff
@@ -246,69 +245,76 @@ def _update_finished_move(self):
+                    move_line.qty_done += abstract_wo.qty_producing
+                else:
+                    location_dest_id = production_move.location_dest_id._get_putaway_strategy(abstract_wo.product_id).id or production_move.location_dest_id.id
+                    move_line_vals.append({
+                        'move_id': production_move.id,
+                        'product_id': production_move.product_id.id,
+                        'lot_id': abstract_wo.finished_lot_id.id,
+                        'product_uom_qty': abstract_wo.qty_producing,
+                        'product_uom_id': abstract_wo.product_uom_id.id,
+                        'qty_done': abstract_wo.qty_producing,
+                        'location_id': production_move.location_id.id,
+                        'location_dest_id': location_dest_id,
+                    })
             else:
-                location_dest_id = production_move.location_dest_id._get_putaway_strategy(self.product_id).id or production_move.location_dest_id.id
-                move_line.create({
-                    'move_id': production_move.id,
-                    'product_id': production_move.product_id.id,
-                    'lot_id': self.finished_lot_id.id,
-                    'product_uom_qty': self.qty_producing,
-                    'product_uom_id': self.product_uom_id.id,
-                    'qty_done': self.qty_producing,
-                    'location_id': production_move.location_id.id,
-                    'location_dest_id': location_dest_id,
-                })
-        else:
-            rounding = production_move.product_uom.rounding
-            production_move._set_quantity_done(
-                float_round(self.qty_producing, precision_rounding=rounding)
-            )
+                rounding = production_move.product_uom.rounding
+                production_move._set_quantity_done(
+                    float_round(abstract_wo.qty_producing, precision_rounding=rounding)
+                )
+        self.env['stock.move.line'].create(move_line_vals)
 
     def _update_moves(self):
         """ Once the production is done. Modify the workorder lines into
         stock move line with the registered lot and quantity done.
         """
         # Before writting produce quantities, we ensure they respect the bom strictness
-        self._strict_consumption_check()
         vals_list = []
-        workorder_lines_to_process = self._workorder_line_ids().filtered(lambda line: line.product_id != self.product_id and line.qty_done > 0)
-        for line in workorder_lines_to_process:
-            line._update_move_lines()
-            if float_compare(line.qty_done, 0, precision_rounding=line.product_uom_id.rounding) > 0:
-                vals_list += line._create_extra_move_lines()
-
-        self._workorder_line_ids().filtered(lambda line: line.product_id != self.product_id).unlink()
+        line_to_unlink = self[:-1]._workorder_line_ids().browse()
```

Comment:
The method won't work if is this call with different abstract workorder (mrp_workorder + mrp_product_produce). It is intentional ?